### PR TITLE
[t5 doc] make the example work out of the box

### LINKED ISSUE #12238: [doc] t5 incomplete example
https://huggingface.co/transformers/model_doc/t5.html#training has:
```
input_ids = tokenizer('translate English to German: The house is wonderful.', return_tensors='pt').input_ids
labels = tokenizer('Das Haus ist wunderbar.', return_tensors='pt').input_ids
# the forward function automatically creates the correct decoder_input_ids
loss = model(input_ids=input_ids, labels=labels).loss
```
which is broken unless the right model type is used. And when none if specified it typically presumes `AutoModel`, which gives: `TypeError: forward() got an unexpected keyword argument 'labels'`

So probably need to use an explicit `T5ForConditionalGeneration` and then it works.

@patrickvonplaten, @patil-suraj

### PR DESCRIPTION
**Pull request by stas00**
This PR expands the training example to include the correct model type for the example to work, e.g. with `T5Model` this example will break.

Fixes: https://github.com/huggingface/transformers/issues/12238

@patrickvonplaten

### CODE REVIEW

File: docs/source/model_doc/t5.rst [hunk 1]
```diff
@@ -87,6 +87,10 @@ token. T5 can be trained / fine-tuned both in a supervised and unsupervised fash
 
 .. code-block::
 
+    from transformers import T5ForConditionalGeneration, T5Tokenizer
+    mname = "t5-small"
+    model = T5ForConditionalGeneration.from_pretrained(mname)
+    tokenizer = T5Tokenizer.from_pretrained(mname)
```
Comment (patil-suraj) [inline docs/source/model_doc/t5.rst:93]:
```suggestion
    from transformers import T5ForConditionalGeneration, T5Tokenizer
    model = T5ForConditionalGeneration.from_pretrained("t5-small")
    tokenizer = T5Tokenizer.from_pretrained(t5-small")
```

Comment (patil-suraj) [reply docs/source/model_doc/t5.rst:93]: Could we also add this in the "Unsupervised denoising training" example?

**Review by patil-suraj — APPROVED** [review]
Thanks for fixing this!

**Review by patrickvonplaten — APPROVED** [review]

File: docs/source/model_doc/t5.rst (hunk 1 above)
Comment (stas00) [reply docs/source/model_doc/t5.rst]: of course, will validate the code and add it.

**Pull request merged**